python/ray/serve/_private/deployment_state.py

```diff
@@ -1595,55 +1578,70 @@ def autoscale(
             current_handle_queued_queries=current_handle_queued_queries,
+            target_capacity=self._target_state.info.target_capacity,
+            target_capacity_direction=self._target_state.info.target_capacity_direction,
         )
 
-        # Clip the replica count by capacity-adjusted bounds.
-        # TODO (shrekris-anyscale): this should logic should be pushed into the
-        # autoscaling_policy. Need to discuss what the right API would look like.
-        upper_bound = self.get_capacity_adjusted_num_replicas(
-            autoscaling_policy.config.max_replicas, target_capacity
-        )
-        if (
-            target_capacity_scale_direction == TargetCapacityScaleDirection.UP
-            and autoscaling_policy.config.initial_replicas is not None
-        ):
-            lower_bound = self.get_capacity_adjusted_num_replicas(
-                autoscaling_policy.config.initial_replicas, target_capacity
-            )
-        else:
-            lower_bound = self.get_capacity_adjusted_num_replicas(
-                autoscaling_policy.config.min_replicas, target_capacity
-            )
-
-        clipped_decision_num_replicas = max(
-            lower_bound, min(decision_num_replicas, upper_bound)
-        )
-
-        if (
-            clipped_decision_num_replicas == self._target_state.num_replicas
-            and self._target_state.adjust_capacity is False
-        ):
+        if decision_num_replicas == self._target_state.target_num_replicas:
             return
 
         logger.info(
             f"Autoscaling replicas for deployment {self.deployment_name} in "
-            f"application {self.app_name} to {clipped_decision_num_replicas}. "
+            f"application {self.app_name} to {decision_num_replicas}. "
             f"current_num_ongoing_requests: {current_num_ongoing_requests}, "
             f"current handle queued queries: {current_handle_queued_queries}."
         )
 
         new_info = copy(self._target_state.info)
-        new_info.set_autoscaled_num_replicas(clipped_decision_num_replicas)
         new_info.version = self._target_state.version.code_version
+
+        has_reached_steady_state = self._has_reached_steady_state()
+        if has_reached_steady_state:
+            new_status_trigger = DeploymentStatusTrigger.AUTOSCALING
+        else:
+            new_status_trigger = self.curr_status_info.status_trigger
+
         self._set_target_state(
             new_info,
-            status_trigger=DeploymentStatusTrigger.AUTOSCALING,
-            adjust_capacity=False,
+            decision_num_replicas,
+            status_trigger=new_status_trigger,
+            allow_scaling_statuses=has_reached_steady_state,
+        )
+
+    def _has_reached_steady_state(self) -> bool:
+        """Whether or not this deployment has reached steady-state.
+
+        This method should only be used for autoscaling deployments. It raises
+        an assertion error otherwise.
```

Comment:
again, don't know what "steady state" is -- this comment does not help me understand what the method is doing in any way

Reply:
I renamed the method and changed the docstring. Let me know if it's clearer.